src/routes/(app)/salto/[door]/+server.ts

```diff
@@ -36,7 +44,36 @@ export const GET: RequestHandler = async ({ locals, params }) => {
     },
   });
 
+  const bannedPositions = await prisma.position.findMany({
+    where: {
+      OR: bannedPositionIds.map((p) => ({
+        id: {
+          startsWith: `${p}%`,
+        },
+      })),
+    },
+  });
+
   // Take positions such as "dsek" and "dsek.km" and return all students with these roles
+  const bannedStudentsFromRoles = (
+    await prisma.mandate.findMany({
+      where: {
+        positionId: {
+          in: bannedPositions.map((p) => p.id),
+        },
+      },
+      select: {
+        member: {
+          select: {
+            studentId: true,
+          },
+        },
+      },
+    })
```

Comment:
I'm pretty sure all of these queries can be made in one (or two, one for accesses and one for bans I guess). It also seems these access policies do not take the current date into account, that's a big problem. But it also seems the old code doesn't check the dates either, which is an even bigger problem (?!)

Reply:
You were right about the old code. It's been fixed on `main`.